scripts/pkg/build_templates/opensearch/rpm/opensearch.rpm.spec

```diff
@@ -89,6 +89,34 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     echo "Stop existing opensearch-performance-analyzer.service"
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
+
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+set +e
+rpm -q opensearch > /dev/null 2>&1 || yum list installed opensearch > /dev/null 2>&1
+EX_CODE=$?
+if [ $EX_CODE -eq 0 ]; then
```

Comment:
`if [ $? -eq 0 ]` here gave me a warning to use the command directly instead of $? comparison. Hence I ended up assigning it to a variable.